src/python/dxpy/__init__.py

```diff
@@ -609,6 +610,12 @@ def ensure_ascii(i):
 
                 response = pool_manager.request(_method, encoded_url, headers=_headers, body=body,
                                                 timeout=timeout, retries=False, **kwargs)
+                # Handle redirection manually for symlink files
+                if response.status in range(300, 399):
+                    redirect_url = response.headers.get('Location')
+                    if not redirect_url:
+                        raise exceptions.UrllibInternalError("Location not found in redirect response", response.status)
+                    break
```

Comment:
sorry just a last minor thing: can you please move this block right below upgrade notify block (line 637 of the updated version)? I've realized it can be handy to set `req_id` first so in case `Location` is missing and exception is thrown so it is reported in error messages and could be used for debugging...